site/content/en/docs/tasks/dev/testing.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Running tests"
+linkTitle: "Running tests"
+weight: 25
+description: >
+  Running tests in Kueue
+---
+
+## Running presubmission verification tests
+```shell
+make verify
+```
+
+## Running unit tests
+To run all unit tests:
+```shell
+make test
+```
+
+To run unit tests for webhooks:
+```shell
+go test ./pkg/webhooks
+```
+To run tests that match `TestValidateClusterQueue` regular expression [ref](https://pkg.go.dev/cmd/go#hdr-Testing_flags):
+```shell
+go test ./pkg/webhooks -run TestValidateClusterQueue
+```
+
+### Running unit tests with race detection
+
+```shell
+# install go stress
+go install golang.org/x/tools/cmd/stress@latest
+# recompile with race instrumentation
+go test ./pkg/scheduler/preemption/ -race -c
+# it loops and reports failures
+stress ./preemption.test -test.run TestPreemption
+```
+
+## Running integration tests
+
+For running subset of tests see [Running subset of tests](#running-subset-of-integration-or-e2e-tests).
+
+### Increase logging verbosity
+You can change log level in the framework (for example, change -3 to -5 to increase verbosity):
+https://github.com/kubernetes-sigs/kueue/blob/f8015cb273f9115c34f9be32b35f7e1308c16459/test/integration/framework/framework.go#L72:
+```go
+var setupLogger = sync.OnceFunc(func() {
+	ctrl.SetLogger(util.NewTestingLogger(ginkgo.GinkgoWriter, -3))
+})
+```
+
+## Running e2e tests using custom build
+```shell
+make kind-image-build test-e2e
+```
+
+For running subset of tests see [Running subset of tests](#running-subset-of-integration-or-e2e-tests).
+
+## Debug tests in VSCode
+It is possible to debug unit and integration tests in VSCode.
+You need to have the [Go extension](https://marketplace.visualstudio.com/items?itemName=golang.Go) installed.
+Now you will have `run test | debug test` text buttons above lines like
+```go
+func TestValidateClusterQueue(t *testing.T) {
+```
+You can click on the `debug test` to debug a specific test.
+
+For integration tests, an additional step is needed.  Run `make test-integration | grep KUBEBUILDER_ASSETS` and add the path to the variable in settings.json:
+```json
+"go.testEnvVars": {
+    "KUBEBUILDER_ASSETS": "<path from output above>",
```

Comment:
If we are testing kueuectl tests, we need to mention that the `KUEUE_BIN` variable must be set.